components/dashboards/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/internal/DashboardRestApi.java

```diff
@@ -309,4 +313,27 @@ private static String replaceCRLFCharacters(String str) {
         }
         return str;
     }
+
+    /**
+     * Gets the configs for pdf
+     *
+     * @return response
+     */
+    @GET
+    @Path("/report-config/{type}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getReportConfigs(@PathParam("type") String type) {
+        try {
+            ReportConfigs configurations = dashboardDataProvider.getReportGenerationConfigurations();
+            Map<String, Object> reportConfigurations = Collections.emptyMap();
+            if (type.equals("pdf")) {
+                reportConfigurations = configurations.getPdf();
+            }
+            return Response.ok().entity(GSON.toJson(reportConfigurations)).build();
```

Comment:
I don't think we need Gson here. You can directly set the `reportConfigurations` object to response.

Reply:
Here the map should be sent as a JSON object, without GSON the data is received as a string not a JSON object.

Reply:
Done